airflow/api_connexion/endpoints/task_instance_endpoint.py

```diff
@@ -73,9 +74,22 @@ def get_task_instance(dag_id: str, dag_run_id: str, task_id: str, session=None):
     return task_instance_schema.dump(task_instance)
 
 
+def _convert_state(states):
+    if not states:
+        return
+    for idx, item in enumerate(states):
+        # we do not support 'null' maybe we should?
+        # to support null, we should add it to TaskState in
+        # openapi doc
+        if item in ['null', 'none']:
+            states[idx] = State.NONE
+    return states
```

Comment:
Instead of modifying the list in place, would it be better to do this instead?

```suggestion
def _convert_state(states):
    if not states:
        return None
    return [State.NONE if item in {"null, "none"} else s for s in states]
```

Also it’d be nice to have type annotation on this function.